server.py

```diff
@@ -113,12 +114,34 @@ def index():
     ), status.HTTP_200_OK
 
 @app.route(PATH_INVENTORY, methods=[GET])
-def list_all_prod_info():
-    """ Return all entries in the Inventory system """
-    all_prod_info = ProductInformation.list_all()
+def query_prod_info():
+    """ Query specific entries in the Inventory system """
+    all_prod_info = []
+    if request.args.get('prod_name'):
+        prod_name = request.args.get('prod_name')
+        if prod_name == None:
```

Comment:
This if block is never processed. If prod_name is None, it wouldn't enter if statement at line 120.